src/Core/Logger.php

```diff
@@ -5,82 +5,333 @@
 namespace Friendica\Core;
 
 use Friendica\BaseObject;
-use Friendica\Core\Config;
-use Friendica\Util\DateTimeFormat;
-use ReflectionClass;
+use Friendica\Network\HTTPException\InternalServerErrorException;
+use Friendica\Util\LoggerFactory;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
 
 /**
  * @brief Logger functions
  */
 class Logger extends BaseObject
 {
-    // Log levels:
-    const WARNING = 0;
-    const INFO = 1;
-    const TRACE = 2;
-    const DEBUG = 3;
-    const DATA = 4;
-    const ALL = 5;
-
-    public static $levels = [
-        self::WARNING => 'Warning',
-        self::INFO => 'Info',
-        self::TRACE => 'Trace',
-        self::DEBUG => 'Debug',
-        self::DATA => 'Data',
-        self::ALL => 'All',
-    ];
+	/**
+	 * @deprecated 2019.03 use Logger::error() instead
+	 * @see Logger::error()
+	 */
+	const WARNING = 0;
```

Comment:
You don't need to keep the former constant value. Since we have been using the constant names everywhere, you can seamlessly change their value without needing to keep the old ones for backwards compatibility.

Reply:
hm .. I think it would be better to have explicit the "old" codes deprectaed so everybody - who has time and find some of this log entries - can replace them the "new" way (with `debug()`, `info()`, .. and `{placeholder}` and the `context` array).

Reply:
Then why not just just deprecate the old `Logger::log` method, and you can change the constant values so that it still is filters correctly, while we replace its instances with `Logger::debug`, `Logger::notice` and `Logger::warning` etc... and a default context can be generated for legacy calls.

Reply:
hm .. sounds good too .. ^^

Reply:
The reason I prefer this is that it allows for a smoother transition with no necessary mapping between legacy constants and PSR-3 ones. We'll just replace logger calls one by one, and there will be less cruft to be removed once we're done migrating every call.